Android/app/src/main/java/com/example/brewhaha_android/Controllers/HomeActivity.kt

```diff
@@ -69,9 +100,69 @@ class HomeActivity(private val api: BackendConnection = BackendConnection()) : A
         }
     }
 
-    fun viewBreweries(tokenBundle: Bundle) {
-        val intent = Intent(baseContext, ViewBreweryActivity::class.java)
-        intent.putExtra("bundle", tokenBundle)
-        startActivity(intent)
+    fun getBreweries() {
+        val token = AuthToken(tokenBundle!!["token"] as String, "", "")
+        Log.d("Home Brewery Call", "getting all breweries")
+        doAsync {
+            val response = api.getAllBreweries(token).execute()
+            if (response.isSuccessful) {
+                breweryList = response.body()
+                Log.d("Home Brewery Call", breweryList.toString())
+                uiThread {
+                    Log.d("Home Brewery Call", "success")
+                    Log.d("Home Brewery Call", "Size: " + breweryList.size)
+                    breweryList.forEach {
+                        Log.d("Home Brewery Call", "Name: " + it)
+                    }
+                }
+
+            }
+            else {
+                uiThread {
+                    Log.d("Home Brewery Call","Code: " + response.code())
+                    Log.d("Home Brewery Call", "Error Message: " + response.errorBody())
+                    Log.d("Home Brewery Call", "Response Body: " + response.message())
+                }
+            }
+        }
+    }
+}
+
+class BreweryAdapter(val breweryList: List<Brewery>): RecyclerView.Adapter<BreweryAdapter.ViewHolder>() {
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        Log.d("Home Adapter", breweryList[position].name)
+        val brewery = breweryList[position]
+        holder.image?.setImageResource(R.drawable.beer_mugs)
+        holder.name?.text = brewery.name
+        holder.address?.text = String.format("%d %s, %s", brewery.address!!.number,
+            brewery.address.line1, brewery.address.postalCode)
+        val rating_double = brewery.friendlinessRating!!.aggregate
+        if (rating_double == null) {
+            holder.rating?.text = "Rating coming soon!"
+        } else {
+            holder.rating?.text = String.format("Rating: %d", rating_double.toString())
+        }
+
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val v = LayoutInflater.from(parent.context).inflate(R.layout.brewery_card, parent, false)
+        return ViewHolder(v)
+    }
+
+
+    override fun getItemCount(): Int {
+        return breweryList.size
+    }
+
+    class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
+
```

Comment:
Should you bind this explicitly to the view its editing (brewery_card)? Might be easier to see what this is actually doing.

Sidenote: how does this work without an explicit setContentView() call?

Reply:
This is how many of the tutorials did it